src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -190,14 +247,46 @@ private static InputValueDefinitionNode GenerateSimpleInputType(NameNode name, F
 
             return new(
                 location: null,
-                field.Name,
-                new StringValueNode($"Input for field {field.Name} on type {inputTypeName}"),
-                type,
+                name: field.Name,
+                description: new StringValueNode($"Input for field {field.Name} on type {inputTypeName}"),
+                type: type,
                 defaultValue: null,
-                field.Directives
+                directives: field.Directives
             );
         }
 
+        /// <summary>
+        /// Helper method to determine if there is a M:N relationship between the parent and child node by checking that the relationship
+        /// directive's cardinality value is Cardinality.Many for both parent -> child and child -> parent.
+        /// </summary>
+        /// <param name="childFieldDefinitionNode">FieldDefinition of the child node.</param>
+        /// <param name="childObjectTypeDefinitionNode">Object definition of the child node.</param>
+        /// <param name="parentNode">Parent node's NameNode.</param>
+        /// <returns>true if the relationship between parent and child entities has a cardinality of M:N.</returns>
+        private static bool IsMToNRelationship(FieldDefinitionNode childFieldDefinitionNode, ObjectTypeDefinitionNode childObjectTypeDefinitionNode, NameNode parentNode)
+        {
+            // Determine the cardinality of the relationship from parent -> child, where parent is the entity present at a level
+            // higher than the child. Eg. For 1:N relationship from parent -> child, the right cardinality is N.
+            Cardinality rightCardinality = RelationshipDirectiveType.Cardinality(childFieldDefinitionNode);
+            if (rightCardinality is not Cardinality.Many)
+            {
+                return false;
+            }
+
+            List<FieldDefinitionNode> fieldsInChildNode = childObjectTypeDefinitionNode.Fields.ToList();
+            int indexOfParentFieldInChildDefinition = fieldsInChildNode.FindIndex(field => field.Type.NamedType().Name.Value.Equals(QueryBuilder.GeneratePaginationTypeName(parentNode.Value)));
```

Comment:
In this check, you are generating the pagination field name ( suffixed with `Connection`) and then checking if there is a field with that name - if there are mappings defined for a field, would it matter at this stage considering the nodes are already created?

Also, it would be helpful to add comments with examples on the check you are performing

Reply:
> if there are mappings defined for a field,

You mean `mappings` section of the entity config? Mappings cannot be defined for relationship entities so we don't need to worry about them here.

Reply:
Outdated code.